tutorials/tensordictmodule.ipynb

```diff
@@ -0,0 +1,931 @@
+    ")\n",
+    "linear = TensorDictModule(nn.Linear(3, 10), in_keys=[\"a\"], out_keys=[\"a_out\"])\n",
+    "linear(tensordict)\n",
+    "assert (tensordict[\"b\"] == torch.zeros(5, 4, 3)).all()\n",
+    "tensordict"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3a818629",
+   "metadata": {},
+   "source": [
+    "We can also do it inplace:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "4d1342d0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tensordict = TensorDict(\n",
+    "    {\"a\": torch.randn(5, 3), \"b\": torch.zeros(5, 4, 3)}, batch_size=[5]\n",
+    ")\n",
+    "\n",
+    "linear = TensorDictModule(nn.Linear(3, 10), in_keys=[\"a\"], out_keys=[\"a\"])\n",
+    "\n",
+    "linear(tensordict)\n",
+    "assert tensordict[\"a\"].shape == torch.Size([5, 10])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "00035cbd",
+   "metadata": {},
+   "source": [
+    "### Example 2: Multiple inputs"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "06a20c22",
+   "metadata": {},
+   "source": [
+    "Now lets imagine a more complex network that takes 2 entries and average them into a single output:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "69098393",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MergeLinear(nn.Module):\n",
+    "    def __init__(self, in_1, in_2, out):\n",
+    "        super().__init__()\n",
+    "        self.linear_1 = nn.Linear(in_1, out)\n",
+    "        self.linear_2 = nn.Linear(in_2, out)\n",
+    "\n",
+    "    def forward(self, x_1, x_2):\n",
+    "        return (self.linear_1(x_1) + self.linear_2(x_2)) / 2"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "2dd686bb",
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "TensorDict(\n",
+       "    fields={\n",
+       "        a: Tensor(torch.Size([5, 3]), dtype=torch.float32),\n",
+       "        b: Tensor(torch.Size([5, 4]), dtype=torch.float32),\n",
+       "        output: Tensor(torch.Size([5, 10]), dtype=torch.float32)},\n",
+       "    batch_size=torch.Size([5]),\n",
+       "    device=cpu,\n",
+       "    is_shared=False)"
+      ]
+     },
+     "execution_count": 6,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "tensordict = TensorDict(\n",
+    "    {\n",
+    "        \"a\": torch.randn(5, 3),\n",
+    "        \"b\": torch.randn(5, 4),\n",
+    "    },\n",
+    "    batch_size=[5],\n",
+    ")\n",
+    "\n",
+    "mergelinear = TensorDictModule(\n",
+    "    MergeLinear(3, 4, 10), in_keys=[\"a\", \"b\"], out_keys=[\"output\"]\n",
+    ")\n",
+    "\n",
+    "mergelinear(tensordict)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "11256ae7",
+   "metadata": {},
+   "source": [
+    "### Example 3: Multiple outputs\n",
+    "We can also map to multiple outputs"
```

Comment:
"`TensorDictModule` not only supports multiple inputs but also multiple outputs"